libfreerdp/core/orders.c

```diff
@@ -2664,7 +2664,7 @@ static BOOL update_decompress_brush(wStream* s, BYTE* output, size_t outSize, BY
 	const BYTE* palette = Stream_Pointer(s) + 16;
 	const size_t bytesPerPixel = ((bpp + 1) / 8);
 
-	if (Stream_GetRemainingLength(s) < 16 + bytesPerPixel * 4)
+	if (Stream_GetRemainingLength(s) < 16ULL + bytesPerPixel * 4ULL)
 		return FALSE;
```

Comment:
Like above to address all the overflow cases, we should probably do a 
`if ((Stream_GetRemainingLength(s) < 16ULL) || ((Stream_GetRemainingLength(s) -16) / 4 < bytesPerPixel)`

Reply:
not required due to range of input data type. `bpp` is (at most) `UINT32` and we operate on `UINT64` here.

Reply:
sorry, even more narrow, at most `UINT8`